src/SBInclinedSersic.cpp

```diff
@@ -0,0 +1,469 @@
+                           _r0 = _info->calculateScaleForTruncatedHLR(_re, _trunc);
+                       }
+
+                       // Update _info with the correct truncated version.
+                       _info = cache.get(boost::make_tuple(_n,_trunc/_r0,
+                                                           this->gsparams.duplicate()));
+
+                       if (flux_untruncated) {
+                           // Update the stored _flux and _re with the correct values
+                           _flux *= _info->getFluxFraction();
+                           _re = _r0 * _info->getHLR();
+                       }
+                   } else {
+                       // Then given HLR and flux are the values for the untruncated profile.
+                       _r0 = _re / _info->getHLR();
+                   }
+               }
+               break;
+          case SCALE_RADIUS:
+               {
+                   _r0 = size;
+                   if (_truncated) {
+                       // Update _info with the correct truncated version.
+                       _info = cache.get(boost::make_tuple(_n,_trunc/_r0,
+                                                           this->gsparams.duplicate()));
+
+                       if (flux_untruncated) {
+                           // Update the stored _flux with the correct value
+                           _flux *= _info->getFluxFraction();
+                       }
+                   }
+                   // In all cases, _re is the real HLR
+                   _re = _r0 * _info->getHLR();
+               }
+               break;
+          default:
+               throw SBError("Unknown SBInclinedSersic::RadiusType");
+        }
+        dbg<<"hlr = "<<_re<<std::endl;
+        dbg<<"r0 = "<<_r0<<std::endl;
+
+        _inv_r0 = 1./_r0;
+        _half_pi_h_sini_over_r = 0.5*M_PI*_h0*std::abs(_inclination.sin())/_r0;
+
+        dbg << "inv_r0 = " << _inv_r0 << std::endl;
+        dbg << "half_pi_h_sini_over_r = " << _half_pi_h_sini_over_r << std::endl;
+
+        _r0_sq = _r0*_r0;
+        _inv_r0 = 1./_r0;
+        _inv_r0_sq = _inv_r0*_inv_r0;
+
+        // Get the inverse exponential HLR, which is used for calculating stepK
+        _inv_exp_re = 1./(1.6783469900166605*_r0);
+
+        _shootnorm = _flux * _info->getXNorm(); // For shooting, we don't need the 1/r0^2 factor.
+        _xnorm = _shootnorm * _inv_r0_sq;
+        dbg<<"norms = "<<_xnorm<<", "<<_shootnorm<<std::endl;
+
+        // For small k, we can use up to quartic in the taylor expansion of both terms
+        // in the calculation.
+        // This is acceptable when the next term is less than kvalue_accuracy.
+        // (35/16 + 31/15120 pi/2*h*sin(i)/r) * (k^2*r^2)^3 = kvalue_accuracy
+        // This is a bit conservative, note, assuming kx = 0
+        _ksq_min = std::pow(this->gsparams->kvalue_accuracy /
+                            (35./16. + 31./15120.*_half_pi_h_sini_over_r), 1./3.);
```

Comment:
Yes, it is. I've put in an adjustment for this now, though it comes with the caveat that it doesn't affect the ksq_min used internally by the SersicInfo class. This isn't much of an issue, as the adjustment needed to it would be quite small (about 1 part in 1000, I believe). It could be fixed, but that would require changes to the SersicInfo class which would mean we'd need to cache it for different half_pi_h_sini_over_r values as well, which is hardly ideal.

One possibility is to just be a little more conservative inside the SersicInfo class and increase its ksq_min slightly to account for this, if it's worth bothering with at all.